### PR TITLE
Migrate project to Swift 4

### PR DESCRIPTION
**Pull request by carol-mohemian**
Resolves migration to Swift 4

### CODE REVIEW

File: Source/SideMenuController+SideOver.swift [hunk 1]
```diff
@@ -141,16 +141,16 @@ extension SideMenuController {
             self.centerPanelOverlay.alpha = alpha
             self.set(statusUnderlayAlpha: alpha)
             self.sidePanel.frame = destinationFrame
-            }, completion: { _ in
+            }, completion: { () in
```
Comment (jordan-pichler) [inline Source/SideMenuController+SideOver.swift]: I think you can leave `() in` out completely in Swift 4 😊

File: Source/SideMenuController+SideUnder.swift [hunk 2]
```diff
@@ -92,18 +92,18 @@ extension SideMenuController {
         
         let updated = centerPanel.frame != centerPanelFrame
         
-        UIView.panelAnimation( duration, animations: { _ in
+        UIView.panelAnimation( duration, animations: { 
             self.centerPanel.frame = centerPanelFrame
             self.set(statusUnderlayAlpha: hidden ? 0 : 1)
-        }) { _ in
+        }) { () in
```
Comment (jordan-pichler) [inline Source/SideMenuController+SideUnder.swift]: same here ^^

File: Example/Example.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -421,7 +434,8 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = com.teodorpatras.sidemenucontroller.example;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_VERSION = 3.0;
+				SWIFT_SWIFT3_OBJC_INFERENCE = On;
```
Comment (jordan-pichler) [inline Example/Example.xcodeproj/project.pbxproj]: Please set ObjC/Swift 3 Inference Behavior to `Default` in the build settings 😉

**Review by jordan-pichler — CHANGES_REQUESTED** [review]
Looking good 👍⭐️ 
Just a few minor thingies and this is good to go! ✏️

File: Source/SideMenuController+SideOver.swift (hunk 1 above)
Comment (carol-mohemian) [reply Source/SideMenuController+SideOver.swift]: Why does it even migrate to this then? 🤔 But yes, you're right.

Comment (jordan-pichler) [reply Source/SideMenuController+SideOver.swift]: I asked myself the same when you noticed all the unnecessary enum prefixes in my PRs 😄

**Review by jordan-pichler — APPROVED** [review]
N🍦 job and you were super fast! 🏎

**Pull request merged**